### PR TITLE
Add tests to hello world example

### PR DESCRIPTION
**Pull request by napcs**
The Hello World example gets pulled in as the default when using the `npx @temporalio/create` script and it's a good starting point for basic Temporal demo apps.

I think it should ship with tests, so I've added three:

1. An Activity test
2. A workflow test that does run the Activity
3. A Workflow test that mocks the Activitiy.

These tests come from the 102 TypeScript curriculum.

They do not include nyc coverage as I thought that complicated things.

Open to feedback.

### CODE REVIEW

**Comment by bergundy** [comment]
@mjameswh want to also take a look?

**Review by angelazhou32 — APPROVED** [review]

File: hello-world/src/mocha/workflows.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { Worker } from '@temporalio/worker';
+import { example } from '../workflows';
+import * as activities from '../activities';
+import assert from 'assert';
+
+describe('Example workflow', () => {
+  let testEnv: TestWorkflowEnvironment;
+
+  before(async () => {
+    testEnv = await TestWorkflowEnvironment.createTimeSkipping();
```
Comment (mjameswh) [inline hello-world/src/mocha/workflows.test.ts:12]: Same

File: hello-world/src/mocha/workflows-mocks.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { after, before, it } from 'mocha';
+import { Worker } from '@temporalio/worker';
+import { example } from '../workflows';
+import assert from 'assert';
+
+describe('Example workflow with mocks', () => {
+  let testEnv: TestWorkflowEnvironment;
+
+  before(async () => {
+    testEnv = await TestWorkflowEnvironment.createTimeSkipping();
```
[the remark this answers is not in the document]
Comment (angelazhou32) [reply hello-world/src/mocha/workflows-mocks.test.ts:11]: Thank you for the reminder!

**Review by mjameswh — APPROVED** [review]
Sorry, had lost trace of this one.

Just two nits. Feel free to merge once that's fixed.

**Comment by mjameswh** [comment]
Thanks @napcs!

**Pull request merged**